### DISCUSSION TITLE
snap7.util.get_real() delivers inaccurate value

### DISCUSSION THREAD

**Discussion by joerggm** [gijzelaerr/python-snap7#371] [Q&A] [answered]
Hello 
I read and write to S7-1500 with no issues. But to extract a REAL value out of a bytearray I get an inaccurate value.
Example:
| S7-DB value| get_real()|
|:-----------------|:---------|
|`0.1` |`0.10000000149011612`|
|`50.0` |`50.0`|
|`50.1` |`50.099998474121094`|
|`49.9` |`49.900001525878906`|

Do i have to accept this or am I doing something wrong? 
I want to save  values in sqlite database.

Thanks in advanced.

**Comment by swamper123** [comment, marked as answer]
You haven't done something wrong. This is the nature of floating point numbers. (
[Siemens said, you can be 6 digits accurate](https://support.industry.siemens.com/cs/document/14844391/how-accurately-can-i-calculate-with-real-numbers-that-are-used-in-extensive-formulas-?dti=0&lc=en-WW) on a PLC.
Then you got snap7 and pyton-snap7 in between and the floatingpoint error exists.

Try out this in your pythonconsole:
```python
>>> 1.2-1.0
0.19999999999999996
>>> 
```
For more infos about this error: [read these python docs](https://docs.python.org/3/tutorial/floatingpoint.html)
